.github/workflows/docstrings_linting.yml

```diff
@@ -0,0 +1,20 @@
+name: run docstrings linting
```

Comment:
Why do we need this? The `pylint` job (we should rename it BTW) already runs `ruff` with the new rules

Reply:
hmmm, I guess it does, but I'm not sure, I would like to check but which other workflow file runs `ruff`? 
I can't seem to find it:

```
/Users/dsbatista/haystack/.github/workflows
(haystack) ~/haystack/.github/workflows [docstrings-linting ⟳ |…1]
09:43 $ grep 'ruff' *
docstrings_linting.yml:      - name: ruff docstrings linting
docstrings_linting.yml:        run: hatch run ruff check haystack
(haystack) ~/haystack/.github/workflows [docstrings-linting ⟳ |…1]
```

Reply:
My bad, hatch runs ruff in the pylint step, but only on the changed files https://github.com/deepset-ai/haystack/actions/runs/8595736108/job/23550964059?pr=7463#step:6:1 sorry for the noise!